pipelines/deployer/deployer.yaml

```diff
@@ -130,6 +130,28 @@ generate_users_terraform: &generate_users_terraform
   outputs:
   - name: users-terraform
 
+generate_namespace_roles_terraform: &generate_namespace_roles_terraform
+  platform: linux
+  run:
+    path: /bin/bash
+    args:
+    - -euo
+    - pipefail
+    - -c
+    - |
+      mkdir -p namespace-roles-terraform
+      cd config
+      echo "creating terraform for namespace roles from cluster config"
+      yq '.namespaces[]' "((config-values-path))" \
```

Comment:
directly placing pipeline variables into scripts is usually frowned upon a bit because

* interpolation of pipeline variables is not always easy to reason about... for example in this case `--check-creds` will not be able to tell you if this variable is unset
* interpolation does not happen if this task was moved into a task _file_ and could be easily missed in a migration
* It is generally harder to read what the task needs if the variables are not decalred in params/vars
* It makes the script invalid to run on its own - this can be important if we were good and actually has tests for our pipeline scripts
* It makes tasks easier to use with `fly execute` for experimenting

The pattern that is used elsewhere in these pipelines is to set things like this via `params` to set environment variables which can be overrided if need be

Reply:
Agreed